test/functional/attributes.js

```diff
@@ -103,6 +103,54 @@ describe('attribute updates', () => {
       expect(el.getAttribute('data-foo')).to.equal(null);
     });
 
+    it('should keep attribute in different position', () => {
```

Comment:
This should be split up into several tests. Also, one of the later tests (dropping foo from the start of an element with 3 attributes) isn't any different than the first test..